src/GafferImage/ImageWriter.cpp

```diff
@@ -1675,59 +1775,150 @@ void ImageWriter::execute() const
 		boost::filesystem::create_directories( directory );
 	}
 
-	if ( out->open( fileName, spec ) )
-	{
-		IECore::msg( IECore::MessageHandler::Info, this->relativeName( this->scriptNode() ), "Writing " + fileName );
-	}
-	else
-	{
-		throw IECore::Exception( boost::str( boost::format( "Could not open \"%s\", error = %s" ) % fileName % out->geterror() ) );
-	}
-
 	// Write out the channel data
 
-	const ColorSpace *appropriateColorSpaceNode = hasAlpha ? colorSpaceUnpremultedNode() : colorSpaceNode();
-
 	const Imath::Box2i extImageDataWindow( Imath::V2i( spec.x, spec.y ), Imath::V2i( spec.x + spec.width - 1, spec.y + spec.height - 1 ) );
 	const Imath::Box2i imageDataWindow( imageFormat.fromEXRSpace( extImageDataWindow ) );
 	const Imath::Box2i processDataWindow( BufferAlgo::intersection( imageDataWindow, dataWindow ) );
 
-	if( !deep )
+	struct Part
 	{
-		TileChannelDataProcessor processor;
+		std::string name;
+		std::string view;
+		std::vector< std::string > channels;
+		std::vector< std::string > channelNames;
+	};
+
+	std::vector< Part > parts;
 
-		if ( spec.tile_width == 0 )
+	std::string currentView = "";
+
+	for( const string &i : channelsToWrite )
+	{
+		auto && [ viewName, partName, channelName ] = viewPartChannelName( this, currentView, i );
```

Comment:
:1st_place_medal:  First use of structured bindings in Gaffer!

What does the `&&` vs `&` buy us in this case?